README.md

```diff
@@ -22,22 +20,109 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+The gem uses the adapter pattern to select the search service you need.
+You can also setup the logger:
+
+
+```
+  SiteSearch.config do |config|
+    config.adapter = :algolia
+    config.logger = Rails.logger
+  end
+```
+
+### Algolia
+
+To do a query you can use the query class.
+An example below search through "retirement" passing algolia specific
+options:
+
+```
+SiteSearch::Query.new(
+  'retirement',
+  options: {
+    index: 'pages',
+    highlightPreTag: '<br>',
+    highlightPostTag: '</br>',
+    page: 1,
+    hitsPerPage: 10
+  }
+).results
+```
+
+### Returning results for all adapters
+
+This will return a collection of SiteSearch::Result:
+
+```
+[
+  #<SiteSearch::Result @title=<br>Retirement</br> income tool,
+    @link=/en/tools/retirement-income-tool, @description=<br>Retirement</br> income
+    tool>
+  #<SiteSearch::Result @title=<br>Retirement</br> adviser directory,
+    @link=/en/tools/retirement-adviser-directory, @description=FCA regulated
+    financial advisers for help with <br>retirement</br> planning, inheritance tax,
+    equity release, wills and probate, long-term care and pension transfers.>
+]
+```
 
 ## Development
 
-After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+### Setup
 
-To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the version number in `version.rb`, and then run `bundle exec rake release`, which will create a git tag for the version, push git commits and tags, and push the `.gem` file to [rubygems.org](https://rubygems.org).
+After checking out the repo, run `bin/setup` to install dependencies.
+
+Then, run `rake spec` to run the tests.
+
+You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+
+To install this gem onto your local machine, run `bundle exec rake install`.
+
+To release a new version, update the version number in `version.rb`,
+and then run `bundle exec rake release`, which will create a git tag for the
+version, push git commits and tags, and push the `.gem` file to [rubygems.org](https://rubygems.org).
+
+The project uses dotenv gem so you can copy the env sample file:
```

Comment:
I would move this above the rake spec line (75). I walked through the setup and hit an error on rake spec before then copying the .env to make it work